main/main.c

```diff
@@ -611,6 +611,7 @@ PHP_INI_BEGIN()
 #ifdef PHP_WIN32
 	STD_PHP_INI_BOOLEAN("windows.show_crt_warning",		"0",		PHP_INI_ALL,		OnUpdateBool,			windows_show_crt_warning,			php_core_globals,	core_globals)
 #endif
+	STD_PHP_INI_ENTRY("syslog.highbit",			"0",		 PHP_INI_SYSTEM,	OnUpdateBool,		syslog_highbit,		php_core_globals, 		core_globals)
```

Comment:
> so what is the other security aspect besides escape sequences? On the other hand, not allowing UTF-8 to pass nowadays is weird.

@weltling Quoting RFC-5424 (which not every syslog server implements, indeed some still implement pre-RFC-3164):

```
6.4.  MSG

   The MSG part contains a free-form message that provides information
   about the event.

   The character set used in MSG SHOULD be UNICODE, encoded using UTF-8
   as specified in [RFC3629].  If the syslog application cannot encode
   the MSG in Unicode, it MAY use any other encoding.

   The syslog application SHOULD avoid octet values below 32 (the
   traditional US-ASCII control character range except DEL).  These
   values are legal, but a syslog application MAY modify these
   characters upon reception.  For example, it might change them into an
   escape sequence (e.g., value 0 may be changed to "\0").  A syslog
   application SHOULD NOT modify any other octet values.

   If a syslog application encodes MSG in UTF-8, the string MUST start
   with the Unicode byte order mask (BOM), which for UTF-8 is ABNF
   %xEF.BB.BF.  The syslog application MUST encode in the "shortest
   form" and MAY use any valid UTF-8 sequence.

   If a syslog application is processing an MSG starting with a BOM and
   the MSG contains UTF-8 that is not shortest form, the MSG MUST NOT be
   interpreted as being encoded in UTF-8, for the reasons outlined in
   [UNICODE-TR36], Section 3.1.  Guidance about this is given in
   Appendix A.8.

   Also, according to UNICODE TR36 [UNICODE-TR36], a syslog application
   MUST NOT interpret messages in the "non-shortest form".  It MUST NOT
   interpret invalid UTF-8 sequences.
```

This wording is problematic, and equivocates a lot.  What does "it MAY use any other encoding" mean?  Does it mean one that uses characters 0x00-0x2f?  Because it goes on to say, "the syslog application SHOULD avoid octet values below 32" which might be seen as a contradiction.

As for, "If a syslog application encodes MSG in UTF-8, the string MUST start with the Unicode byte order mask (BOM), which for UTF-8 is ABNF %xEF.BB.BF." -- do we check the message for that now?  Or simply prepend it?

And as for, "If a syslog application is processing an MSG starting with a BOM and the MSG contains UTF-8 that is not shortest form, the MSG MUST NOT be interpreted as being encoded in UTF-8 [...]", do we need to parse every message for UTF-8 shortest-form conformance?  Is there a library to do that?  Is it already part of the PHP runtime?

Do we accept multibyte strings and transcode them from UTF-32 or UTF-16 into UTF-8?  Why not use UTF-7?  Then we don't have to worry about high-bit characters or control characters... Just exceeding the UDP MTU (usually 1500 bytes for remote logging)... or that some syslog servers have a 1K maximum message size.

There are a lot of unknowns.  Maybe it's simplest to start with a point fix, which is what I was doing, and then gradually add functionality as it's proven to be both safe _and_ correct.